Sources/Apodini/Request/Request.swift

```diff
@@ -5,28 +5,69 @@
 //  Created by Paul Schmiedmayer on 7/12/20.
 //
 
-import Vapor
+import Foundation
+import NIO
+@_implementationOnly import Vapor
 
+/// A type representing a server response.
+public protocol Request {
+    /// parameters
+    func parameter<T: Decodable>(for id: String) throws -> T
+    /// body
+    func bodyParameter<T: Decodable>() throws -> T?
+    /// body data
+    var bodyData: Data? { get }
+    /// The EventLoop associated with this request
+    var eventLoop: EventLoop { get }
+    /// endpoint
+    var endpoint: String? { get }
+    /// database
+    var database: Any? { get }
+}
+
+typealias TotallyNotVaporRequest = Vapor.Request
+
+extension TotallyNotVaporRequest: Request {
+    public func parameter<T: Decodable>(for id: String) throws -> T {
+        try self.query.get(at: id)
+    }
+
+    public func bodyParameter<T: Decodable>() throws -> T? {
```

Comment:
Do we really need this function? It will only work for JSON encoded body-data, thus the user could not rely on it with every request's context.

Reply:
For REST I think we are only dealing with JSON data anyway. If you want to use this for gRPC you might have to implement your own Request / Response types handling the decoding/encoding in a different way.

Reply:
Ah, I see. I think I misunderstood parts of your approach.
So by having a specific `Request` implementation for different bodies, having the decoding in the `SemanticModelBuilder`s is unnecessary 👍 Sorry, my bad.
But I am not sure how we would decode multiple `@Parameter`s for Request content (like described [here](https://github.com/Apodini/Apodini/blob/develop/Documentation/PropertyWrapper/Parameter.md#multiple-parameters-for-request-content)) with the current `bodyParameter`. I assume every `@Parameter` of type `.body` would try to call `request.bodyParameter()` in their `inject` method. But in such a case, the `T` type in `bodyParameter` would not match the entire body, but rather one (of the multiple) nested elements contained in the body.

Reply:
I still need to find a way to map a Parameters UUID to the right Parameter in Vapor.Request.